### PR TITLE
sync shop between connected clients

### LINKED ISSUE #279: [BUG] Shops do not update when multiple players are observing them
**Describe the bug**
When two players open a shop, the shop interface does not properly update when a purchase or sale is made to the shop. The other player is not made aware of it, and can still buy the amount of items that they see in the shop, even if it doesn't reflect the actual amount available to purchase.

**To Reproduce**
1. Log into two accounts
2. Open the same shop
3. Buy or sell an item

**Expected behavior**
Both observers should instantly see the changes made to the Subject ItemContainer.

### PR DESCRIPTION
**Pull request by Promises**
fixes #279

### CODE REVIEW

File: src/game-engine/config/shop-config.ts [hunk 1]
```diff
@@ -91,6 +97,12 @@ export class Shop {
 
     public open(player: Player): void {
         player.metadata['lastOpenedShop'] = this;
+        player.metadata['shopCloseListener'] = player.interfaceState.closed.subscribe((whatClosed: WidgetClosedEvent) => {
+            if(whatClosed && whatClosed.widget && whatClosed.widget.widgetId === widgets.shop.widgetId) {
+                this.removePlayerFromShop(player);
+            }
+        })
```
Comment (Tynarus) [inline src/game-engine/config/shop-config.ts]: semicolon

File: src/game-engine/config/shop-config.ts [hunk 2]
```diff
@@ -103,8 +115,26 @@ export class Shop {
             slot: 'tabarea',
             multi: true
         });
+        this.customers.push(player);
     }
 
+    private updateCustomers() {
+        for (const player of this.customers) {
+            if(player.metadata['lastOpenedShop'] === this){
```
Comment (Tynarus) [inline src/game-engine/config/shop-config.ts]: space between `){` pls

**Review by Tynarus — CHANGES_REQUESTED** [review]
Two tiny future-lint related requests

**Comment by Promises** [comment]
fixed those linting errors you found

**Review by Tynarus — APPROVED** [review]

**Pull request merged**